### PR TITLE
PLDM Support for MEX FRU & MEX PDR

### PR DESCRIPTION
**Pull request by ArchanaKakani**
This PR Includes
- MEX FRU & MEX LED Support
- MEX CM Support

Tested By :

1. On a rainier system with a MEX drawer attached, power on , power off and reset reload works fine.
2. Toggling LED for FAN works find

### CODE REVIEW

File: host-bmc/host_pdr_handler.cpp [hunk 1]
```diff
@@ -935,7 +1026,7 @@ void HostPDRHandler::_setHostSensorState()
                 sensorRearm.byte = 0;
                 uint8_t tid = std::get<0>(terminusInfo);
 
-                auto instanceId = instanceIdDb.next(mctp_eid);
+                auto instanceId = instanceIdDb.next(mctpEid);
```
Comment (Pavithrab7) [inline host-bmc/host_pdr_handler.cpp:1029]: please free the same mctpEid in line 1039

File: host-bmc/host_pdr_handler.cpp [hunk 2]
```diff
@@ -1239,6 +1331,150 @@ void HostPDRHandler::getFRURecordTableByRemote(uint16_t& totalTableRecords)
...
+            if (this->isHostOff)
+            {
+                error("Host is off, stopped sending sensor states command");
+                return;
+            }
+
+            ++sensorMapIndex;
+            if (sensorMapIndex == sensorMap.end())
+            {
+                // error("sensor map completed");
```
Comment (Pavithrab7) [inline host-bmc/host_pdr_handler.cpp:1390]: do we need this?

File: host-bmc/host_pdr_handler.cpp [hunk 3]
```diff
@@ -1239,6 +1331,150 @@ void HostPDRHandler::getFRURecordTableByRemote(uint16_t& totalTableRecords)
...
+                    associations{{ledFwdAssociation, ledReverseAssociation,
+                                  ledGroupPath}};
+                CustomDBus::getCustomDBus().setAssociations(path, associations);
+            }
+        }
+
+        ++sensorMapIndex;
+        if (sensorMapIndex == sensorMap.end())
+        {
+            // error("sensor map completed");
```
Comment (Pavithrab7) [inline host-bmc/host_pdr_handler.cpp:1459]: here also

File: host-bmc/host_pdr_handler.cpp [hunk 4]
```diff
@@ -1288,14 +1524,74 @@ void HostPDRHandler::setFRUDataOnDBus(
                     }
                 }
             }
+            else
+#endif
```
Comment (Pavithrab7) [inline host-bmc/host_pdr_handler.cpp:1528]: can this endif be before else?

**Review by Pavithrab7 — CHANGES_REQUESTED** [review]

File: host-bmc/host_pdr_handler.cpp (hunk 4 above)
Comment (ArchanaKakani) [reply host-bmc/host_pdr_handler.cpp:1526]: If we put endif before else compilation will fail if oem-ibm is not defined.

File: host-bmc/host_pdr_handler.cpp (hunk 3 above)
Comment (ArchanaKakani) [reply host-bmc/host_pdr_handler.cpp]: Removed

File: host-bmc/host_pdr_handler.cpp (hunk 2 above)
Comment (ArchanaKakani) [reply host-bmc/host_pdr_handler.cpp]: Removed

File: host-bmc/host_pdr_handler.cpp (hunk 1 above)
Comment (ArchanaKakani) [reply host-bmc/host_pdr_handler.cpp:1029]: done

**Review by Pavithrab7 — APPROVED** [review]

**Review by manojkiraneda — APPROVED** [review]

**Pull request merged**